dexter/src/main/java/com/karumi/dexter/DexterInstance.java

```diff
@@ -166,7 +166,7 @@ void onCancelPermissionRequest() {
   private void requestPermissionsToSystem(Collection<String> permissions) {
     if (!isShowingNativeDialog.get()) {
       androidPermissionService.requestPermissions(activity,
-          permissions.toArray(new String[permissions.size()]), PERMISSIONS_REQUEST_CODE);
+          permissions.toArray(new String[0]), PERMISSIONS_REQUEST_CODE);
```

Comment:
Why did you change this? If the array is not initialized with the expected size the code will not work.

Reply:
No the code wont break. This change was actually suggested by the IDEA. Numerous sources even claim that using array size 0 is faster than creating an array of the list size manually:

https://medium.com/@hanru.yeh/collection-toarray-new-t-0-will-be-faster-than-collection-toarray-new-t-size-5b129fabfa88

https://stackoverflow.com/questions/174093/toarraynew-myclass0-or-toarraynew-myclassmylist-size

Reply:
> There are two styles to convert a collection to an array: either using a pre-sized array (like c.toArray(new String[c.size()])) or using an empty array (like c.toArray(new String[0]).
> 
> In older Java versions using pre-sized array was recommended, as the reflection call which is necessary to create an array of proper size was quite slow. However since late updates of OpenJDK 6 this call was intrinsified, making the performance of the empty array version the same and sometimes even better, compared to the pre-sized version. Also passing pre-sized array is dangerous for a concurrent or synchronized collection as a data race is possible between the size and toArray call which may result in extra nulls at the end of the array, if the collection was concurrently shrunk during the operation.
> 
> This inspection allows to follow the uniform style: either using an empty array (which is recommended in modern Java) or using a pre-sized array (which might be faster in older Java versions or non-HotSpot based JVMs).

https://github.com/JetBrains/intellij-community/blob/master/plugins/InspectionGadgets/src/inspectionDescriptions/ToArrayCallWithZeroLengthArrayArgument.html

Reply:
Interesting!